packages/semantic-tokens/scripts/tokenGen.ts

```diff
@@ -0,0 +1,380 @@
+/*
+ * Token generation script
+ * Takes in a Figma token export file and generates token raw strings and CSS Var files
+ */
+import tokensJSONRaw from './tokens.json';
+import { fluentOverrides as fluentFallbacksRaw, FluentOverrideValue, type FluentOverrides } from './fluentOverrides';
+import fs from 'fs';
+import { Project } from 'ts-morph';
+import { format } from 'prettier';
+
+const project = new Project({
+  tsConfigFilePath: './tsconfig.json',
+});
+
+function chopLastCamelCasePart(str: string) {
+  const parts = str.split(/(?=[A-Z])/);
+  parts.pop();
+  return parts.join('');
+}
```

Comment:
Is it supposed to chop last part even if it has a single camel case? If not, consider rewriting to something like this: 

```ts
const chopLastCamelCasePart = (str: string) => str.replace(/([a-z])([A-Z])/g, '$1 $2').split(' ').slice(0, -1).join('');
```
this should guarantee only the last camel case is omitted